contracts/jobs/JobsManager.sol

```diff
@@ -70,94 +163,300 @@ contract JobsManager is IJobsManager, Verifiable, Controllable {
      */
     function job(string _streamId, string _transcodingOptions, uint256 _maxPricePerSegment) external returns (bool) {
         address electedTranscoder = bondingManager().electActiveTranscoder(_maxPricePerSegment);
+        /* There must be an elected transcoder */
+        require(electedTranscoder != address(0));
+
+        Job storage job = jobs[numJobs];
+        job.jobId = numJobs;
+        job.streamId = _streamId;
+        job.transcodingOptions = _transcodingOptions;
+        job.maxPricePerSegment = _maxPricePerSegment;
+        job.broadcasterAddress = msg.sender;
+        job.transcoderAddress = electedTranscoder;
+
+        NewJob(electedTranscoder, msg.sender, numJobs);
+
+        // Increment number of created jobs
+        numJobs = numJobs.add(1);
+
+        return true;
+    }
+
+    /*
+     * @dev End a job. Can be called by either a broadcaster or transcoder of a job
+     * @param _jobId Job identifier
+     */
+    function endJob(uint256 _jobId) external returns (bool) {
+        Job storage job = jobs[_jobId];
+
+        // Job must not already have an end block
+        require(job.endBlock == 0);
+        // Sender must be the job's broadcaster or elected transcoder
+        require(job.broadcasterAddress == msg.sender || jobs[_jobId].transcoderAddress == msg.sender);
+
+        // Set end block for job
+        job.endBlock = block.number.add(jobEndingPeriod);
+    }
+
+    /*
+     * @dev Submit claim for a range of segments
+     * @param _jobId Job identifier
+     * @param _segmentRange Range of claimed segments
+     * @param _claimRoot Merkle root of transcoded segment proof data for claimed segments
+     */
+    function claimWork(uint256 _jobId, uint256[2] _segmentRange, bytes32 _claimRoot) jobExists(_jobId) external returns (bool) {
+        Job storage job = jobs[_jobId];
+
+        // Job must be active
+        require(jobStatus(_jobId) == JobStatus.Active);
+        // Sender must be elected transcoder
+        require(job.transcoderAddress == msg.sender);
+
+        // Move fees from broadcaster deposit to escrow
+        uint256 fees = _segmentRange[1].sub(_segmentRange[0]).add(1).mul(job.maxPricePerSegment);
```

Comment:
Yeah I agree that the price should be `transcoder.rates.pricePerSegment` instead of `job.maxPricePerSegment` the rationale being that a transcoder can set the lowest price per segment it is willing to accept and if a job's max price per segment is higher than that, the transcoder would still be happy to accept the job at the price it originally set.